### ISSUE TITLE
Crashing when marking blocks as code

### ISSUE THREAD

**Issue by koitsu** [SourMesen/Mesen#507]
Report is for Mesen 0.9.7 on Windows 7 Ultimate SP1 x64 with .NET 

I've been working on RE'ing a particular MMC1 game.

Several times (four, to be exact) in the past few hours (see screenshot), I've been getting app-not-responding/hangs or actual crashes when trying to mark unidentified blocks as code while in the debugger (i.e. Debugger -> navigating around until I find an "unidentified block" that certainly is code -> right-clicking on a row and choosing Mark Selection As... -> Verified Code).  I've also had this happen when marking things as code inside of the Memory Viewer in the same fashion.

The last/most recent incident actually resulted in an exception with a call backtrace (see screenshot), though the window became grey due the nature of Windows (heh ;-) ), but I did manage to snag a screenshot of the dialog in case it's of any help.

I went as far as to delete my entire `Debugger\` directory just to ensure I started from a clean slate; no difference.

The issue definitely relates to "marking" things as code vs. data vs. unidentified, as it only seem to happen then.

Some options I've changed that might be relevant, but unsure.  Maybe having the game actively running while in the debugger doing the above might somehow trigger a particular quirk/issue:

* Debugger -> Options -> Refresh UI while running **checked**
* Debugger -> Options -> Break Options -> Break on debugger focus **unchecked**

Happy to provide `settings.xml` if needed.  I'll do my best to assist debugging this in any way possible, including running debug builds if asked.  I've been trying to get exact reproduction steps for you, but it's tricky keeping track of *everything* I do (every click, navigation, etc.).  I could try to record a video of a session if that would be helpful.

If you need to give me a debug build or something you don't want to link publicly, just hit me up privately on the nesdev forum.

**Comment by koitsu** [comment]
![ss00](https://user-images.githubusercontent.com/8430773/46072865-484aef80-c138-11e8-8727-e654822dcd63.png)

![ss01](https://user-images.githubusercontent.com/8430773/46072874-4bde7680-c138-11e8-9595-9e6bde41a63a.png)

**Comment by koitsu** [comment]
For the 3 actual crashes, I managed to find the individual `.dmp` files Windows generated.  I can put these up somewhere if asked (they're about 90MBytes each, but all 3 .zip up into 15MB).

Using Microsoft's Debug Diagnostic Tool, I can get some further data, but honestly the call trace in the screenshot seems most useful considering there's no debug symbols in MesenCore.dll.  Here's a summary of the function call stack of all 3 crash incidents:

01:26:27 incident (this one sucked, because I'd had Mesen open for over 2 hours):
```
Mesen.GUI.InterOpEmu.Run()
Mesen.GUI.InterOpEmu.Run()

MesenCore!DebugInitialize+8041
MesenCore!blip_add_delta+ac74
MesenCore!blip_add_delta+1b0c5
MesenCore!Run+15
```

02:29:48 incident (Mesen only running for about 1 minute):
```
Mesen.GUI.InterOpEmu.Run()
Mesen.GUI.InterOpEmu.Run()

MesenCore!blip_add_delta+ac50
MesenCore!blip_add_delta+1b0c5
MesenCore!Run+15
```

02:35:05 incident (Mesen running for about 4.5 minutes):
```
Mesen.GUI.InterOpEmu.Run()
Mesen.GUI.InterOpEmu.Run()

MesenCore!DebugInitialize+8041
MesenCore!blip_add_delta+ac74
MesenCore!blip_add_delta+1b0c5
MesenCore!Run+15
```

**Comment by SourMesen** [comment]
Thanks for the report!
I'm fairly sure this should be fixed by 9e5c22492b38f2da8d5975040266b55fbde524f4 (managed to reproduce the crash once without that fix, and never crashed after the fix)

There's a dev build with the fix on appveyor here: https://ci.appveyor.com/project/Sour/mesen/build/artifacts  (Direct link: https://ci.appveyor.com/api/buildjobs/d48arc7labxfmc9c/artifacts/MesenDevWin-0.9.7.3.exe)

If you're still getting the crash with this build, let me know and I'll take another look.

**Comment by koitsu** [comment]
Been doing a lot of RE recently (hours upon hours) using the above build; appears to be fixed!  Thanks much (and enjoy your vacation!)
